internal/pkg/term/selector/selector.go

```diff
@@ -196,6 +203,131 @@ type WorkspaceSelector struct {
+	workingDirAbs, err := os.Getwd()
+	if err != nil {
+		return nil, fmt.Errorf("get working directory: %w", err)
+	}
+	return &localFileSelector{
+		prompt:        prompt,
+		fs:            &afero.Afero{Fs: fs},
+		workingDirAbs: workingDirAbs,
+	}, nil
+}
+
+// Dockerfile asks the user to select from a list of Dockerfiles in the current
+// directory or one level down. If no dockerfiles are found, it asks for a custom path.
+func (s *localFileSelector) Dockerfile(selPrompt, notFoundPrompt, selHelp, notFoundHelp string, pathValidator prompt.ValidatorFunc) (string, error) {
+	dockerfiles, err := s.listDockerfiles()
+	if err != nil {
+		return "", err
+	}
+	var sel string
+	dockerfiles = append(dockerfiles, []string{dockerfilePromptUseCustom, DockerfilePromptUseImage}...)
+	sel, err = s.prompt.SelectOne(
+		selPrompt,
+		selHelp,
+		dockerfiles,
+		prompt.WithFinalMessage(dockerfileFinalMsg),
+	)
+	if err != nil {
+		return "", fmt.Errorf("select Dockerfile: %w", err)
+	}
+	if sel != dockerfilePromptUseCustom {
+		return sel, nil
+	}
+	sel, err = s.prompt.Get(
+		notFoundPrompt,
+		notFoundHelp,
+		pathValidator,
+		prompt.WithFinalMessage(dockerfileFinalMsg))
+	if err != nil {
+		return "", fmt.Errorf("get custom Dockerfile path: %w", err)
+	}
+	return sel, nil
+}
+
+// ListDockerfiles returns the list of Dockerfiles within the current
+// working directory and a sub-directory level below. If an error occurs while
+// reading directories, or no Dockerfiles found returns the error.
+func (s *localFileSelector) listDockerfiles() ([]string, error) {
+	wdFiles, err := s.fs.ReadDir(s.workingDirAbs)
+	if err != nil {
+		return nil, fmt.Errorf("read directory: %w", err)
+	}
+	var dockerfiles = make([]string, 0)
+	for _, wdFile := range wdFiles {
+		// Add current file if it is a Dockerfile and not a directory; otherwise continue.
+		if !wdFile.IsDir() {
+			fname := wdFile.Name()
+			if strings.Contains(strings.ToLower(fname), dockerfileName) && !strings.HasSuffix(strings.ToLower(fname), dockerignoreName) {
+				path := filepath.Dir(fname) + "/" + fname
+				dockerfiles = append(dockerfiles, path)
+			}
+			continue
+		}
+
+		// Add sub-directories containing a Dockerfile one level below current directory.
+		subFiles, err := s.fs.ReadDir(wdFile.Name())
+		if err != nil {
+			// swallow errors for unreadable directories
+			continue
+		}
+		for _, f := range subFiles {
+			// NOTE: ignore directories in sub-directories.
+			if f.IsDir() {
+				continue
+			}
+			fname := f.Name()
+			if strings.Contains(strings.ToLower(fname), dockerfileName) && !strings.HasSuffix(strings.ToLower(fname), dockerignoreName) {
+				path := wdFile.Name() + "/" + f.Name()
+				dockerfiles = append(dockerfiles, path)
+			}
+		}
+	}
+	sort.Strings(dockerfiles)
+	return dockerfiles, nil
+}
+
+type configurationSelector struct {
+	prompt prompter
+}
+
+// NewConfigurationSelector constructs a localFconfigurationSelectorileSelector.
```

Comment:
nit: there is a copy/pasta typo here
```suggestion
// NewConfigurationSelector constructs a configurationSelector.
```